src/cpu.jakt

```diff
@@ -1419,13 +1419,29 @@ class CPU {
 
     function branch(mut this) {
         let arg_address = .pc + 1
-        let address: i8 = .system.read_byte(address: arg_address) as! i8
-
+        let operand: i8 = .system.read_byte(address: arg_address) as! i8
+        
+        // The 6502 would have the PC on the next opcode now and everything is
+        // based upon that. Lets emulate that which also makes the code look cleaner
+        .pc += 2
+        
+        let offset: u8 = 0
+        let negative = false
```

Comment:
does this ever change? I see it checked here but then doesn't seem to get updated

Reply:
You are totally right, I forgot to update it whenever the offset was negative.
Should all be fixed now.